update/v1alpha1/types_update_status.go

```diff
@@ -0,0 +1,750 @@
+}
+
+// UpdateStatusSpec is empty for now, can possibly hold configuration for Update Status Controller in the future
+type UpdateStatusSpec struct {
+}
+
+// +k8s:deepcopy-gen=true
+
+// UpdateStatusStatus is the API about in-progress updates, kept populated by Update Status Controller by
+// aggregating and summarizing UpdateInsights produced by update informers
+type UpdateStatusStatus struct {
+	// controlPlane contains a summary and insights related to the control plane update
+	ControlPlane ControlPlaneUpdateStatus `json:"controlPlane"`
+
+	// workerPools contains summaries and insights related to the worker pools update
+	WorkerPools []PoolUpdateStatus `json:"workerPools"`
+
+	// conditions provide details about Update Status Controller operational matters
+	// +listType=map
+	// +listMapKey=type
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// ControlPlaneConditionType are types of conditions that can be reported on control plane level
+type ControlPlaneConditionType string
+
+const (
+	// ControlPlaneUpdating is the condition type that communicate whether the whole control plane is updating
+	// or not
+	ControlPlaneUpdating ControlPlaneConditionType = "Updating"
+)
+
+// ControlPlaneUpdatingReason are well-known reasons for the Updating condition
+// +kubebuilder:validation:Enum=ClusterVersionProgressing;ClusterVersionNotProgressing;CannotDetermineUpdating
+type ControlPlaneUpdatingReason string
+
+const (
+	// ReasonClusterVersionProgressing is used for Updating=True set because we observed a ClusterVersion resource to
+	// have Progressing=True condition
+	ReasonClusterVersionProgressing ControlPlaneUpdatingReason = "ClusterVersionProgressing"
+	// ReasonClusterVersionNotProgressing is used for Updating=False set because we observed a ClusterVersion resource to
+	// have Progressing=False condition
+	ReasonClusterVersionNotProgressing ControlPlaneUpdatingReason = "ClusterVersionNotProgressing"
+	// ReasonClusterVersionCannotDetermine is used for Updating=Unknown. This covers many different actual reasons such as
+	// missing or Unknown Progressing condition on ClusterVersion, but it does not seem useful to track the individual
+	// reasons to that granularity for Updating=Unknown
+	ReasonClusterVersionCannotDetermine ControlPlaneUpdatingReason = "CannotDetermineUpdating"
+)
+
+// ControlPlaneUpdateStatus contains a summary and insights related to the control plane update
+type ControlPlaneUpdateStatus struct {
+	// resource is the resource that represents the control plane. It will typically be a ClusterVersion resource
+	// in standalone OpenShift and HostedCluster in HCP.
+	// +required
+	Resource ResourceRef `json:"resource"`
+
+	// poolResource is the resource that represents control plane node pool, typically a MachineConfigPool. This field
+	// is optional because some form factors (like HCP) do not have dedicated control plane node pools.
```

Comment:
nit: "HCP" -> "hosted control planes", so folks don't have to be familiar enough in this space to be able to unpack acronyms themselves.